src/workerd/api/node/buffer.c++

```diff
@@ -597,5 +606,211 @@ uint32_t BufferUtil::write(
   return writeInto(js, buffer, string, offset, length, getEncoding(encoding));
 }
 
+// ======================================================================================
+// StringDecoder
+
```

Comment:
maybe add some documentation on decoder state? Seems to be an array with multiple elements? [missing, buffered,...]?